File: ds/ds_mid/MySet.java
public class MySet extends ArrayList<Object>{
	//This code does not work, but it illustrates the idea
	public boolean member(Object o){
		//Check if o is in set
		for(Object elem : this){
			if(elem.equals(o)){
				return true;
			}
		}
		return false;
	}

	public MySet insert(Object o){
		//Insert object o into set
		this.add(o);
	}

	public MySet remove(int n){
		//Remove object at index n from set
		this.remove(n);
	}		
}

File: ds/ds_mid/MySet.hs
data Set a = Empty | Elem a (Set a)
	deriving (Show)

member :: Eq a => a -> Set a -> Bool
member a Empty = False
member a (Elem b (rest)) = if a == b then True else member a rest

insert :: Eq a => a -> Set a -> Set a
insert a Empty = Elem a (Empty)
insert a set   = if member a set then set else Elem a (set)

remove :: Eq a => a -> Set a -> Set a
remove a Empty = Empty
remove a set = if member a set then remove' a set else set
	where
		remove' a (Elem b (rest)) = if a == b then rest else Elem b (remove' a rest)
